test/end-to-end/cases-coverage/xsl-choose-01.xsl

```diff
@@ -31,29 +31,85 @@
         <xsl:otherwise version="3.0">                                          <!-- Expected miss -->
           <node type="choose otherwise" version="3.0">                         <!-- Expected miss -->
-            <xsl:text version="3.0">200</xsl:text>                             <!-- Expected miss -->
+            <xsl:text version="3.0">0</xsl:text>                               <!-- Expected miss -->
           </node>                                                              <!-- Expected miss -->
         </xsl:otherwise>                                                       <!-- Expected miss -->
       </xsl:choose>
-      <!-- xsl:when doesn't match so when not executed. otherwise has no content. -->
+      <!-- xsl:when not executed. xsl:otherwise executes but has no content, so unknown. -->
       <xsl:choose>
         <xsl:when test="1 eq 2" version="3.0">                                 <!-- Expected miss -->
           <node type="choose when" version="3.0">                              <!-- Expected miss -->
-            <xsl:text version="3.0">100</xsl:text>                             <!-- Expected miss -->
+            <xsl:text version="3.0">0</xsl:text>                               <!-- Expected miss -->
           </node>                                                              <!-- Expected miss -->
         </xsl:when>                                                            <!-- Expected miss -->
-        <xsl:otherwise></xsl:otherwise>
+        <xsl:otherwise></xsl:otherwise>                                        <!-- Expected unknown -->
       </xsl:choose>
+      <!-- xsl:when not executed, but status is unknown. -->
+      <xsl:choose>
+        <xsl:when test="exists(nonexistent)" version="3.0"></xsl:when>         <!-- Expected unknown -->
+        <xsl:when test="exists(missing)" version="3.0">                        <!-- Expected unknown -->
+          <!--untraced node-->                                                 <!-- Expected ignored -->
+        </xsl:when>                                                            <!-- Expected unknown -->
+        <xsl:when test="exists(omitted)" version="3.0">                        <!-- Expected unknown -->
+          <xsl:where-populated>                                                <!-- Expected unknown -->
+          </xsl:where-populated>                                               <!-- Expected unknown -->
+        </xsl:when>                                                            <!-- Expected unknown -->
+      </xsl:choose>
+      <!-- xsl:when executes, but status is unknown. -->
+      <!-- xsl:otherwise not executed, but status is unknown. -->
+      <xsl:for-each select="(1,2)">
+        <node type="choose when executed unknown">
+          <xsl:choose>
+            <xsl:when test=". eq 1" version="3.0">                             <!-- Expected unknown -->
```

Comment:
@birdya22, I wasn't sure of the reason for the `version="3.0"` attributes that were already in this file, though my guess is that it is related to [4.0 changes](https://qt4cg.org/specifications/xslt-40/Overview.html#xsl-choose). Can you check if I used that attribute correctly in the material that I added here?

Reply:
Annoyingly I don't seem to have made any note of why they are there. 
I think they are related to issue [1920](https://github.com/xspec/xspec/issues/1920) where I said:

> The basic problem is that if the parent element of a text() node includes any attributes then xspec reports a miss on that text() node.

In my test case for 1920 I generally used `expand-text="no"` to get an attribute on an element. I guess I went for something with zero impact when doing the `xsl:choose` text case, `version="3.0"` (not sure why I only did it on the `xsl:choose` text case. 
I don't think they should be there now because they were trying to test something different from the `xsl:choose` test case, so I would suggest removing them all.
With the change to use the Linked tree ([pull request 1936](https://github.com/xspec/xspec/pull/1936)) the #1920 issue has gone away. 
Actually with 12.5 using the Tiny tree the issue has also gone away because of the changes to reported column numbers for text nodes.

Reply:
Thanks for explaining, as I hadn't made that connection. With 78a11e3d260645a6ebaca74c2c6701f69e3fa75e, I removed all the version attributes in that file, except on `xsl:stylesheet`.